File: ruby/rapleaf_example.rb
require 'rapleaf_api'

=begin
  This example script takes an e-mail as a command line argument 
	and queries Rapleaf's database for any data associated with
	the provided e-mail (unknown fields are left blank)
=end

email = ARGV.first
begin
  response = RapleafApi.query_by_email(email)
  if response.empty?
    puts "No Data Found"
  else
    response.each do |k,v|
      puts "#{k} = #{v}"
    end
  end
rescue Exception => e
  puts e.message
end

File: python/rapleafApi/RapleafExample.py
import RapleafApi
import sys

""" This example script takes an e-mail as a command line argument 
and queries Rapleaf's database for any data associated with
the provided e-mail (unknown fields are left blank) 
The hash returned from query_by_email is iterated through
and each k/v pair is sent to std out """
  
api_key = 'SET_ME'        # Set your API key here
email = sys.argv[1]         # Command line argument
api = RapleafApi.RapleafApi(api_key)   # Instance of the API class
try:
    response = api.query_by_email(email)
    for k, v in response.iteritems():
        print('%s = %s' % (k, v))
except Exception as inst:       # HTTP code returned other than 200
    code, body = inst.args
    print('Error Code = %d' % code)
    print('Error Body = %s' % body)
